ci/build-src-tarball.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -ex
+
+# Determine the name of the tarball
+tag=dev
+if [[ $GITHUB_REF == refs/tags/v* ]]; then
+  tag=${GITHUB_REF:10}
+fi
+pkgname=wasmtime-$tag-src
+rm -rf /tmp/$pkgname
+mkdir /tmp/$pkgname
+
+# Vendor all crates.io dependencies since this is supposed to be an
+# offline-only-compatible tarball
+mkdir /tmp/$pkgname/.cargo
+cargo vendor > /tmp/$pkgname/.cargo/config.toml
+
+# Copy over everything in-tree to the destination
+cp -r * /tmp/$pkgname
```

Comment:
Neat! I had to use `--exclude=.git` instead of `--exclude-vcs` because otherwise it was ignoring files in vendored directories which Cargo didn't like, but I prefer to avoid copying things around.